File: benchmark/output/python-pygame/openai/breakout_game.py
import pygame
import sys

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Breakout")
clock = pygame.time.Clock()

# Game state variables
paddle_x = 360.0
paddle_speed = 300.0

ball_x, ball_y = 400.0, 400.0
ball_vx, ball_vy = 200.0, -200.0
ball_radius = 6

brick_alive = [1.0] * 8
brick_width, brick_height = 90, 20
brick_spacing = 100
bricks_y = 50

score = 0
game_over = False

font = pygame.font.Font(None, 36)

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    keys = pygame.key.get_pressed()
    if keys[pygame.K_LEFT]:
        paddle_x -= paddle_speed * dt
    if keys[pygame.K_RIGHT]:
        paddle_x += paddle_speed * dt

    # Keep paddle within screen bounds
    paddle_x = max(0, min(paddle_x, 720))

    # Update ball position
    ball_x += ball_vx * dt
    ball_y += ball_vy * dt

    # Ball collision with walls
    if ball_x < ball_radius or ball_x > 800 - ball_radius:
        ball_vx = -ball_vx
    if ball_y < ball_radius:
        ball_vy = -ball_vy

    # Ball collision with paddle
    if ball_y >= 564 and paddle_x <= ball_x <= paddle_x + 80:
        ball_vy = -ball_vy

    # Ball collision with bricks
    for i in range(8):
        if brick_alive[i]:
            brick_x = i * brick_spacing + 5
            if brick_x <= ball_x <= brick_x + brick_width and bricks_y <= ball_y <= bricks_y + brick_height:
                brick_alive[i] = 0.0
                ball_vy = -ball_vy
                score += 1

    # Check if ball is below the screen
    if ball_y > 600:
        game_over = True

    # Draw everything
    screen.fill((0, 0, 0))
    pygame.draw.rect(screen, (255, 255, 255), (int(paddle_x), 570, 80, 12))
    pygame.draw.circle(screen, (255, 255, 255), (int(ball_x), int(ball_y)), ball_radius)

    for i in range(8):
        if brick_alive[i]:
            brick_x = i * brick_spacing + 5
            pygame.draw.rect(screen, (255, 255, 255), (brick_x, bricks_y, brick_width, brick_height))

    score_text = font.render(f"Score: {score}", True, (255, 255, 255))
    screen.blit(score_text, (10, 10))

    if game_over:
        game_over_text = font.render("GAME OVER", True, (255, 255, 255))
        screen.blit(game_over_text, (350, 300))

    pygame.display.flip()

pygame.quit()
sys.exit()

File: vibe-lang/benchmark/output/lua-love/openai/breakout_game.lua
local paddle_x = 360
local paddle_speed = 300
local ball_x = 400
local ball_y = 400
local ball_radius = 6
local ball_vx = 200
local ball_vy = -200
local brick_width = 90
local brick_height = 20
local brick_alive = {1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0}
local score = 0
local game_over = false

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Breakout")
end

function love.update(dt)
    if love.keyboard.isDown("left") then
        paddle_x = paddle_x - paddle_speed * dt
    end
    if love.keyboard.isDown("right") then
        paddle_x = paddle_x + paddle_speed * dt
    end

    ball_x = ball_x + ball_vx * dt
    ball_y = ball_y + ball_vy * dt

    if ball_x < ball_radius or ball_x > 800 - ball_radius then
        ball_vx = -ball_vx
    end
    if ball_y < ball_radius then
        ball_vy = -ball_vy
    end

    if ball_y >= 564 and ball_x >= paddle_x and ball_x <= paddle_x + 80 then
        ball_vy = -ball_vy
    end

    for i = 0, 7 do
        if brick_alive[i + 1] == 1.0 then
            local brick_x = i * 100 + 5
            if ball_x > brick_x and ball_x < brick_x + brick_width and ball_y > 50 and ball_y < 50 + brick_height then
                brick_alive[i + 1] = 0.0
                ball_vy = -ball_vy
                score = score + 1
            end
        end
    end

    if ball_y > 600 then
        game_over = true
    end
end

function love.draw()
    love.graphics.rectangle("fill", paddle_x, 570, 80, 12)
    love.graphics.circle("fill", ball_x, ball_y, ball_radius)

    for i = 0, 7 do
        if brick_alive[i + 1] == 1.0 then
            local brick_x = i * 100 + 5
            love.graphics.rectangle("fill", brick_x, 50, brick_width, brick_height)
        end
    end

    love.graphics.print("Score: " .. score, 10, 10)

    if game_over then
        love.graphics.print("GAME OVER", 350, 300)
    end
end